content/hoon/lagoon/README.md

```diff
@@ -0,0 +1,146 @@
+---
+description: "Documentation for the %base desk's /lib/lagoon.hoon library, which contains linear algebra operations."
+layout:
+  title:
+    visible: true
+  description:
+    visible: false
+  tableOfContents:
+    visible: true
+  outline:
+    visible: true
+  pagination:
+    visible: true
+---
+
+# Lagoon (Linear AlGebra in hOON)
+
+Documentation for the `%base` desk's `/lib/lagoon.hoon` library, which contains linear algebra operations.
+
+Lagoon is a standard library which lives outside of the `/sys` kernel and supports userspace applications which need vectors, matrices, and other linear algebra data structures and operators.  Lagoon is not currently subject to kelvin versioning.
+
+Lagoon is intended to support general-purpose linear algebraic types, but at this point it focuses on integers and IEEE 754 floating-point numbers.
+
+## Data Types
+
+Lagoon's data types are supplied by `/sur/lagoon`.
+
+```
+/-  lagoon
+```
+
+## `$ray`
+
+$n$-dimensional data array
+
+A pair of a bare array as an LSB atom with an associated metadata descriptor.
+
+#### Source
+
+```
++$  ray               ::  $ray:  n-dimensional array
+  $:  =meta           ::  descriptor
+      data=@ux        ::  data, row-major order, 1-pin MSB
+  ==
+```
+
+#### Discussion
+
+## `$meta`
+
+Metadata for `$ray`
+
+A description of the necessary metadata to disambiguate an atom into an array.  The dimensionality of the array is derived from `shape`; the block width from `bloq`; and the type of data each entry should be interpreted as from `kind`.  An arbitrary noun `tail` is supplied for various current and future purposes.
```

Comment:
Should be `.shape`, `.bloq`, `.kind`, `.tail` etc. as per https://docs.urbit.org/style.

Reply:
Same probably applies to any text in backticks that isn't prepended by some kind of symbol.